inventory-management/tests/api.test.js

```diff
@@ -0,0 +1,168 @@
+                "pid": 1,
+                "pcode": "ABC123",
+                "price": 42.5,
+                "sku": "XYZ",
+                "amount_in_stock": 10,
+                "pname": "Something",
+                "description": "This is something very interesting that you want to buy.",
+                "lang": "en_US"
+            },
+            {
+                "pid": 2,
+                "pcode": "FD2",
+                "price": 99.99,
+                "sku": "QWOP",
+                "amount_in_stock": 100,
+                "pname": "Speedos",
+                "description": "I'm too lazy to write a description",
+                "lang": "en_US"
+            }
+        ]
+
+        // Create the records in the database
+        console.log(`Inserting sample records`);
+        await inventory.deleteAll(); // Clear out the existing data
+
+        // Kind of a hack to access the other tables, replace this with models eventually
+        await inventory.repository.knex('products').insert(sample_products);
+    }
+
+    async function initModels() {
+        try {
+            await require('../src/configs/config');
+
+            // Set up all the required constants
+            const { Model } = require('../src/models/inventory-management/index');
+            return { inventory: new Model() }
+        }
+        catch (err) {
+            console.log(err.message);
+            throw err;
+        }
+    }
+
+    // Runs before all tests
+    before(function before() {
+
+        // Load the config and wait for it to load
+        initModels().then(objs => {
+            console.log("my my1");
+            inventory = objs.inventory;
+
+            // Log the start of the test
+            console.log(`Starting test at ${new Date().toLocaleString()}`);
+
+            // Load the sample data into the database
+            loadSampleData();
+
+            product = {
+                "pid": 3,
+                "pcode": "PQR123",
+                "price": 100,
+                "sku": "LMN",
+                "amount_in_stock": 20,
+                "pname": "Watch",
+                "description": "This is a Tissot classic watch.",
+                "lang": "en_US"
+            };
+
+            // Choose a sample product id
+            productId = 3;
+        })
+    })
+
+    // Before each test, clear out the data
+    beforeEach(async function beforeEach() {
+        await inventory.deleteAll();
+    })
+
+    // Test API functionality
+    it("lists products", async () => {
+        // Check if product listing is possible
+        const res = await requestInventory.get('').expect(200);
+        expect(res.body.data).to.eql([{
+            products:
+                '(1,ABC123,42.5,XYZ,10,Something,"This is something very interesting that you want to buy.",en_US)'
+        },
+        {
+            products:
+                '(2,FD2,99.99,QWOP,100,Speedos,"I\'m too lazy to write a description",en_US)'
+        }]);
+    });
```

Comment:
This test is failing for me. Output:

```
1) Inventory management REST API
lists products:

AssertionError: expected [] to deeply equal [ Array(2) ]
+ expected - actual

-[]
+[
+  {
+    "products": "(1,ABC123,42.5,XYZ,10,Something,\"This is something very interesting that you want to buy.\",en_US)"
+  }
+  {
+    "products": "(2,FD2,99.99,QWOP,100,Speedos,\"I'm too lazy to write a description\",en_US)"
+  }
+]

at Context.it (tests/api.test.js:100:34)
  at <anonymous>
at process._tickCallback (internal/process/next_tick.js:188:7)
```

Reply:
hmm..this is weird.. cause I just pulled the repo and ran the setup and the tests.. and they passed

Reply:
@thezeroalpha Can you try again? They must run. If there is an error do let me know

Reply:
OK if it works for both of you then it's probably fine.